phpunit/html/wp-html-tag-processor-test.php

```diff
@@ -297,23 +297,26 @@ public function test_set_attribute_prevents_xss( $attribute_value ) {
 		preg_match( '~^<div test=(.*)></div>$~', (string) $p, $match );
 		list( , $actual_value ) = $match;
 
-		$this->assertEquals( $actual_value, '"' . esc_attr( $attribute_value ) . '"' );
+		$this->assertEquals( $actual_value, '"' . $expected_result . '"' );
 	}
 
 	/**
 	 * Data provider with HTML attribute values that might need escaping.
 	 */
 	public function data_set_attribute_escapable_values() {
 		return array(
-			array( '"' ),
-			array( '&quot;' ),
-			array( '&' ),
-			array( '&amp;' ),
-			array( '&euro;' ),
-			array( "'" ),
-			array( '<>' ),
-			array( '&quot";' ),
-			array( '" onclick="alert(\'1\');"><span onclick=""></span><script>alert("1")</script>' ),
+			array( '"', '&quot;' ),
+			array( '&quot;', '&quot;' ),
+			array( '&', '&amp;' ),
+			array( '&amp;', '&amp;' ),
+			array( '&euro;', '&euro;' ),
+			array( "'", '&#039;' ),
+			array( '<>', '&lt;&gt;' ),
+			array( '&quot";', '&amp;quot&quot;;' ),
+			array(
+				'" onclick="alert(\'1\');"><span onclick=""></span><script>alert("1")</script>',
+				'&quot; onclick=&quot;alert(&#039;1&#039;);&quot;&gt;&lt;span onclick=&quot;&quot;&gt;&lt;/span&gt;&lt;script&gt;alert(&quot;1&quot;)&lt;/script&gt;',
+			),
```

Comment:
@dmsnell I agree – I only did that to have an automated check in place to confirm that `esc_attr` is indeed used there.  I guess we could just provide a spy implementation that always returns a string like `ESC_ATTR( $original_input )` and test for that.

Reply:
well blimey, we can do this with the `attribute_escape` filter

Reply:
added in 80452d8da340f59a5b77f24d7f76825888b34370
should be easy to revert if we need to.
I removed the comment at the top about running this in isolation since we can't anymore; not if we want to assert that we called `esc_attr()` and also verify behaviors of `esc_attr()`. this is fine.

Reply:
Thank you! Looks good to me, too. This PR needs a rebase but should be good otherwise